packages/devtools-solana/test/transactions/signer.test.ts

```diff
@@ -0,0 +1,103 @@
+import fc from 'fast-check'
+import { Connection, sendAndConfirmTransaction, SystemProgram, Transaction } from '@solana/web3.js'
+import { serializeTransactionMessage, OmniSignerSolana, deserializeTransactionBuffer } from '@/transactions'
+import { keypairArbitrary, solanaAddressArbitrary } from '@layerzerolabs/test-devtools-solana'
+import { OmniTransaction } from '@layerzerolabs/devtools'
+import { endpointArbitrary } from '@layerzerolabs/test-devtools'
+
+jest.mock('@solana/web3.js', () => {
+    const original = jest.requireActual('@solana/web3.js')
+
+    return {
+        ...original,
+        sendAndConfirmTransaction: jest.fn(),
+    }
+})
+
+const sendAndConfirmTransactionMock = sendAndConfirmTransaction as jest.Mock
+
+describe('transactions/signer', () => {
+    beforeEach(() => {
+        sendAndConfirmTransactionMock.mockReset()
+    })
+
+    describe('sign', () => {
+        it('should sign a transaction', async () => {
+            await fc.assert(
+                fc.asyncProperty(
+                    endpointArbitrary,
+                    solanaAddressArbitrary,
+                    keypairArbitrary,
+                    keypairArbitrary,
+                    async (eid, address, sender, recipient) => {
```

Comment:
`fast-check` is a pretty cool framework! Looks like it lets us mocks parameters (kind of like solidity fuzz testing parameters)